src/types/composer.ts

```diff
@@ -0,0 +1,673 @@
+  onComplete?: algosdk.OnApplicationComplete
+  /** ID of the application */
+  appId?: bigint
+  /** The program to execute for all OnCompletes other than ClearState */
+  approvalProgram?: Uint8Array
+  /** The program to execute for ClearState OnComplete */
+  clearProgram?: Uint8Array
+  /** The state schema for the app. This is immutable. */
+  schema?: {
+    /** The number of integers saved in global state */
+    globalUints: number
+    /** The number of byte slices saved in global state */
+    globalByteSlices: number
+    /** The number of integers saved in local state */
+    localUints: number
+    /** The number of byte slices saved in local state */
+    localByteSlices: number
+  }
+  /** Application arguments */
+  args?: Uint8Array[]
+  /** Account references */
+  accountReferences?: string[]
+  /** App references */
+  appReferences?: bigint[]
+  /** Asset references */
+  assetReferences?: bigint[]
+  /** Number of extra pages required for the programs */
+  extraPages?: number
+  /** Box references */
+  boxReferences?: algosdk.BoxReference[]
+}
+
+export type MethodCallParams = CommonTxnParams &
+  Omit<AppCallParams, 'args'> & {
+    /** ID of the application */
+    appId: bigint
+    /** The ABI method to call */
+    method: algosdk.ABIMethod
+    /** Arguments to the ABI method */
+    args?: (algosdk.ABIValue | Txn)[]
+  }
+
+type Txn =
+  | (PayTxnParams & { type: 'pay' })
+  | (AssetCreateParams & { type: 'assetCreate' })
+  | (AssetConfigParams & { type: 'assetConfig' })
+  | (AssetFreezeParams & { type: 'assetFreeze' })
+  | (AssetDestroyParams & { type: 'assetDestroy' })
+  | (AssetTransferParams & { type: 'assetTransfer' })
+  | (AssetOptInParams & { type: 'assetOptIn' })
+  | (AppCallParams & { type: 'appCall' })
+  | (KeyRegParams & { type: 'keyReg' })
+  | (algosdk.TransactionWithSigner & { type: 'txnWithSigner' })
+  | { atc: algosdk.AtomicTransactionComposer; type: 'atc' }
+  | (MethodCallParams & { type: 'methodCall' })
+
+export default class AlgokitComposer {
+  /** Map of txid to ABI method */
+  private txnMethodMap: Map<string, algosdk.ABIMethod> = new Map()
+
+  /** Transactions that have not yet been composed */
+  private txns: Txn[] = []
+
+  /** The underlying AtomicTransactionComposer */
+  atc: algosdk.AtomicTransactionComposer
+
+  /** The algod client used by the composer for suggestedParams */
+  algod: algosdk.Algodv2
+
+  /** An async function that will return suggestedParams */
+  getSuggestedParams: () => Promise<algosdk.SuggestedParams>
+
+  /** A function that takes in an address and return a signer function for that address */
+  getSigner: (address: string) => algosdk.TransactionSigner
+
+  /** The default transaction validity window */
+  defaultValidityWindow = 10
+
+  constructor({
+    algod,
+    getSigner,
+    getSuggestedParams,
+    defaultValidityWindow,
+  }: {
+    algod: algosdk.Algodv2
```

Comment:
Maybe we can use a named type here for the constructor args so we can add comments to the various properties

Reply:
Done in a94cf05e8db519f01a34c2c5e07efcb66e3010cb